packages/workspace/src/workspace/reference_indexes.ts

```diff
@@ -42,15 +41,33 @@ import { RemoteMap, RemoteMapEntry } from './remote_map'
 const log = logger(module)
 const { awu } = collections.asynciterable
 
-export const REFERENCE_INDEXES_VERSION = 9
+export const REFERENCE_INDEXES_VERSION = 10
 export const REFERENCE_INDEXES_KEY = 'reference_indexes'
 
 type ChangeReferences = {
   removed: ReferenceInfo[]
   currentAndNew: ReferenceInfo[]
 }
 
-export type ReferenceTargetIndexValue = collections.treeMap.TreeMap<{ id: ElemID; type: ReferenceType }>
+export type ReferenceIndexEntry = {
+  id: ElemID
+} & Pick<ReferenceInfo, 'type' | 'sourceScope'>
+
+export type SerializedReferenceIndexEntry = {
+  id: string
+} & Pick<ReferenceInfo, 'type' | 'sourceScope'>
+
+const isValidReferenceSourceScope = (value: unknown): value is ReferenceInfo['sourceScope'] =>
```

Comment:
since that type is exported anyway, might as well use it directly.
It would also make it easier to find this place if anyone ever changes to possible values in `ReferenceSourceScope`
```suggestion
const isValidReferenceSourceScope = (value: unknown): value is ReferenceSourceScope =>
```

This is still not ideal because there is nothing enforcing that all the possible values are caught here AFAICT

Reply:
Changed the code so adding new values won't break :)